qrenderdoc/Windows/Dialogs/CaptureDialog.cpp

```diff
@@ -434,7 +531,17 @@ void CaptureDialog::on_exePathBrowse_clicked()
   }
 
   if(!filename.isEmpty())
+  {
     SetExecutableFilename(filename);
+
+    if(m_Ctx.Replay().CurrentRemote() &&
+       m_Ctx.Replay().CurrentRemote()->Hostname.startsWith(lit("adb:")))
+    {
+      QByteArray hostnameBytes = m_Ctx.Replay().CurrentRemote()->Hostname.toUtf8();
+      ui->androidLayerWarn->setVisible(
+          !RENDERDOC_CheckAndroidLayer(hostnameBytes.data(), filename.toStdString().c_str()));
```

Comment:
Yeah, especially as it gets more expensive to do the check (i.e. permissions).  I'll make it asynchronous and clean up the string handling.